src/cmd/scan_e2e_test.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2025 Defense Unicorns
+// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/defenseunicorns/uds-pk/src/scan"
+	"github.com/google/go-github/v73/github"
+)
+
+func simulateGrype(args []string, stdout io.Writer, stderr io.Writer) {
+	if stdout == nil {
+		stdout = io.Discard
+	}
+	if stderr == nil {
+		stderr = io.Discard
+	}
+	sep := -1
+	for i, a := range args {
```

Comment:
Why are you manually parsing flags if you could try re-using `flag` package?

Reply:
because I don't know of `flag` package existence :)